### ISSUE TITLE
Error Message Displayed: Failed to switch back to Final Cut Pro when Batch exporting active timeline 

### ISSUE THREAD

**Issue by estudioek** [fcpxhacks/fcpxhacks#3378]
Hey there, First of all thank you for such a great plugin, But I have been using command post for the last couple of years and just since I have updated to FCPX 10.8.1 I can no longer Batch  export  active timeline. A year or to ago I had a similar problem with command post just failing but has been woking just fine for the last year or so with FCPX 10.7.1 and actually an older version of command post 1.4.24 but I just started using FCPX 18.1 3 days ago and got the cannot switch back to FCPX error so I updated to the latest command post and same thing. If I try to export more than 2-3 clips it happens every single time. 
I use your app on daily basis and really am going to suffer with out it working and actually do not even know any other way to bath export clips other than the making compound clip "trick" but I need to export at least 60-100 clips and usually do it with command post in batches.
Thank you so much if you have any suggestions I might be able to try or if you all are aware of this problem.
By the way I am on Ventura 13.6.4, could that maybe be part of the problem? DO I need to update to Sonoma for any reason?
I will do what ever it takes LOL I really need to get this working!!
Thank you very much

**Comment by latenitefilms** [comment]
Will investigate. But for now, I'd suggest just using Frame.io?

**Comment by estudioek** [comment]
Thanks, but I do not use Frame.io I love Command post and will patiently wait to see if you can work this out...
I appreciate it. 
Command post is hands down the best!

**Comment by estudioek** [comment]
Can you all please try an address this or maybe give a reason as to why this may happen? Why would it fail to switch back to FCPX while exporting the time line clips? 
Thanks

**Comment by latenitefilms** [comment]
Normally it's because Final Cut Pro is struggling with too many exports on slower computers. I'm guessing you'd run into the same issue if you tried to do things manually.

**Comment by estudioek** [comment]
> Normally it's because Final Cut Pro is struggling with too many exports on slower computers. I'm guessing you'd run into the same issue if you tried to do things manually.

NO, not the problem bc I have a Studio Max M2 with plenty of Power my friend

**Comment by skymakai** [comment]
I'm having a somewhat similar issue with Batch Export, though on FCPX 11.  My hardware is also quite adequate, being an M2 Ultra, 64GB RAM and plenty of space on the SSD.  

Batch export starts fine and seems to successfully queue up 4-5 clips for export.  After that, it seems to slow down substantially.  It doesn't look as though there's any renders happening.  It seems as though Commandpost itself is simply stuck.  Eventually, it queues up another clip for rendering.  But for more than 4-5 clips it's nearly unusable.  I have to give it an hour to export 30-40 clips.

**Comment by estudioek** [comment]
> I'm having a somewhat similar issue with Batch Export, though on FCPX 11. My hardware is also quite adequate, being an M2 Ultra, 64GB RAM and plenty of space on the SSD.
> 
> Batch export starts fine and seems to successfully queue up 4-5 clips for export. After that, it seems to slow down substantially. It doesn't look as though there's any renders happening. It seems as though Commandpost itself is simply stuck. Eventually, it queues up another clip for rendering. But for more than 4-5 clips it's nearly unusable. I have to give it an hour to export 30-40 clips.

Mine is finally working again after upgrading my os to Sonoma , but export timeline never worked for me if I selected more than a certain amount of clips from the timeline. Not sure what the “limit” is but it was always not more than 10-20 clips max and depends how long the clip is. Even now that it is working again if a choose too many clips , it just doesn’t even start exporting them. So just manually select 10 or so clips , export them then 10 more etc etc until the the whole timeline is exported. Hope that helps. Sure would be nice to be able to export the whole timeline but that is not the case and I cannot complain bc it is the only way to export individual clips at the moment and is a lifesaver for me.

**Comment by Adamki2000** [comment]
I'd also like to chime in and say that the Batch Export Active Timeline almost never seems to work right for me. I have an M1 Max system with the latest Sequoia and FCPX updates. Usually I can't even get Batch Export to work correctly for 2-4 clips, let alone 10 or more. It's incredibly slow to export after even just the first clip, and half the time it just stalls out and never finishes at all. This shouldn't be a hard feature to implement, and it seems like it's a pretty critical one for a lot of people. I'm keeping fingers crossed that this bug or group of bugs gets fixed real soon.
